metaflow/package.py

```diff
@@ -75,6 +76,20 @@ def path_tuples(self):
                 addl_suffixes=self.metaflow_extensions_addl_suffixes,
             ):
                 yield path_tuple
+
+        # Any custom packages exposed via decorators
+        deco_module_paths = set()
+        for step in self._flow:
+            for deco in step.decorators:
+                for path_tuple in deco.add_to_package():
+                    file_path, file_name = path_tuple
+                    # Check if the path is not duplicated as
+                    # many steps can have the same packages being imported
+                    check_name = "%s-%s" % (file_path, file_name)
```

Comment:
Maybe I am being a bit too pedantic here but I would actually do something like this:
```
check_dict = {}
if file_name not in check_dict or check_dict[file_name] == file_path:
  check_dict[file_name] = file_path
  yield path_tuple
elif file_name in check_dict:
  # Raise some error saying that we already included a file for that path.

Reply:
Yes this change seems much better than what I committed but I have a small alteration to propose to it:

```python
if file_name not in deco_module_paths:
    deco_module_paths[file_name] = file_path
    yield path_tuple
elif deco_module_paths[file_name] == file_path:
    pass
elif file_name in deco_module_paths: # this means non unique file_name to file_path mapping. 
    raise NonUniqueFileNameToFilePathMappingException(
        file_name, [deco_module_paths[file_name],file_path]
    )
```
With the change you suggested we `yield path_tuple` even when `check_dict[file_name] == file_path` which we don't want to do to avoid deduplication; Does that make sense ? 

Also Piggbacking to what you said :

It also seems necessary to raise an error for this as it violates the uniqueness constraint of `file_name` to `file_path` mapping. There can be scenarios that people have a python package of `metaflow-cards-scorecard` installed via pip and the same is also present in the python path. In such cases we should fail with and error that `file_name` related to the following `file_path`s and needs to only have a unique `file_name` to `file_path` mapping. 

Just a QQ: 
What should be a name for this exception (Naming things is hard 😄 ) I have for now called it `NonUniqueFileNameToFilePathMappingException`